File: dotnet/Multiplayer/Players/Ban/Ban.cs
using Michitai.Multiplayer;

namespace Michitai.Multiplayer.Players.Ban
{
    /// <summary>
    /// Static class providing utility methods for ban-related operations.
    /// </summary>
    public static class Ban
    {
        /// <summary>
        /// Checks if an API response indicates the player is banned.
        /// </summary>
        /// <param name="response">The API response to check.</param>
        /// <returns>True if the error message indicates the player is banned, false otherwise.</returns>
        public static bool IsBanned(ApiResponse response)
        {
            if (!response.Success && response.Error != null)
            {
                return response.Error.Contains("You are banned");
            }
            return false;
        }
    }
}

File: java/src/main/java/com/michitai/multiplayer/players/ban/Ban.java
package com.michitai.multiplayer.players.ban;

import com.michitai.multiplayer.ApiResponse;

/**
 * Utility methods for ban-related operations.
 */
public final class Ban {
    private Ban() {
    }

    /**
     * Checks if an API response indicates the player is banned.
     *
     * @param response The API response to check.
     * @return True if the error message indicates the player is banned, false otherwise.
     */
    public static boolean isBanned(ApiResponse response) {
        return response != null
            && !response.isSuccess()
            && response.getError() != null
            && response.getError().contains("You are banned");
    }
}
